lib/buildConnectorsJson/generateSchemaFromJs.js

```diff
@@ -5,129 +5,106 @@
 var _            = require('lodash');
 var prettyTitle  = require('./prettyTitle');
 
-
-module.exports = function (config) {
-	return generateSchema(config, true);
+//Standard supported keys
+var supportedKeys = [
+	'type',
+	'title',
+	'enum',
+	'oneOf',
+	'name',
+	'default',
+	'description',
+	'lookup',
+	'format'
+];
+
+module.exports = function (rawSchema) {
+	var schema = generateSchema(rawSchema, true);
+	//Since $schema is only required at root level, assign it here.
+	schema['$schema'] = "http://json-schema.org/draft-04/schema#";
+	 // Standard restriction: top level schema never allows for additional properties
+	schema['additionalProperties'] = false;
+	return schema;
 };
 
-// Separate function down here so it can recursively call itself
-function generateSchema(config, topLevel) {
+//Generates the whole schema for a given  (but not necessarily the schema of a object value)
+function generateSchema(rawSchema, topLevel) {
 
 	var schema = {
-    type: "object",
-    properties: {}
+	    type: "object",
+	    properties: {},
+		required: []
 	};
 
-	if (topLevel === true) {
-		schema.$schema = "http://json-schema.org/draft-04/schema#";
-	}
-
-
-	var inputKeys = _.keys(config);
+	//Advanced only need for top level
+	if (topLevel) schema.advanced = [];
```

Comment:
Hmm I disagree with this. We only use advanced properties on the top level right now, but there's no reason why we couldn't/shouldn't add them on the sub level. @alirussell what do you think?